handler.py

```diff
@@ -129,33 +138,36 @@ def postprocess_and_write(result_dict):
     :param result_dict: dictionary containing all required opts for each example
     """
     _thr = [0.38, 0.13, 0.14]
-    pred_coefs = [1.0] * 4 # not 12, b/c already took mean over 3 in each subset 
-    loc_coefs = [1.0] * 4 
+    pred_coefs = [1.0] * 4  # not 12, b/c already took mean over 3 in each subset
+    loc_coefs = [1.0] * 4
 
     preds = []
     _i = -1
-    for k,v in result_dict.items():
+    for k, v in result_dict.items():
         if 'cls' in k:
             _i += 1
             # Todo: I think the below can just be replaced by v['cls'] -- should check
             msk = v['cls'].numpy()
             preds.append(msk * pred_coefs[_i])
-    
+
     preds = np.asarray(preds).astype('float').sum(axis=0) / np.sum(pred_coefs) / 255
-    
-    loc_preds = []
-    _i = -1
-    for k,v in result_dict.items():
-        if 'loc' in k:
-            _i += 1
-            msk = v['loc'].numpy()
-            loc_preds.append(msk * loc_coefs[_i])
-    
-    loc_preds = np.asarray(loc_preds).astype('float').sum(axis=0) / np.sum(loc_coefs) / 255
-    
     msk_dmg = preds[..., 1:].argmax(axis=2) + 1
-    msk_loc = (1 * ((loc_preds > _thr[0]) | ((loc_preds > _thr[1]) & (msk_dmg > 1) & (msk_dmg < 4)) | ((loc_preds > _thr[2]) & (msk_dmg > 1)))).astype('uint8')
-    
+
+    if v['poly_chip'] == 'None':  # Must be a string or PyTorch throws an error
+        loc_preds = []
+        _i = -1
+        for k, v in result_dict.items():
+            if 'loc' in k:
+                _i += 1
+                msk = v['loc'].numpy()
+                loc_preds.append(msk * loc_coefs[_i])
+
+        loc_preds = np.asarray(loc_preds).astype('float').sum(axis=0) / np.sum(loc_coefs) / 255
+        msk_loc = (1 * ((loc_preds > _thr[0]) | ((loc_preds > _thr[1]) & (msk_dmg > 1) & (msk_dmg < 4)) | (
```

Comment:
This code is so hairy. Can we add in some comments for documentation?

Reply:
Nope...well, maybe you or @jdunnmon can.